File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {number[]}
 */
var kWeakestRows = function(mat, k) {
  const mati = mat.map((r, i) => [r, i])
  const hash = (r, i) => 100 * (r.reduce((a, b) => a + b)) + i
  mati.sort(([r1, i1], [r2, i2]) => hash(r1, i1) - hash(r2, i2))
  return mati.map(([r, i]) => i).slice(0, k)
};

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.go
func sum(row []int) int {
  ans := 0
  for _, v := range row {
    ans += v
  }
  return ans
}

func kWeakestRows(mat [][]int, k int) []int {
  m := len(mat)
  hash := make([]int, 0, m)
  for i, r := range mat {
    hash = append(hash, sum(r) * 100 + i)
  }
  sort.Ints(hash)
  for i := 0; i < k; i ++ {
    hash[i] %= 100
  }
  return hash[:k]
}